velox/dwio/text/tests/writer/TextWriterTest.cpp

```diff
@@ -82,74 +106,104 @@ TEST_F(TextWriterTest, write) {
   const auto filename = "test_text_writer.txt";
   auto filePath = fs::path(fmt::format("{}/{}", tempPath, filename));
   auto sink = std::make_unique<dwio::common::LocalFileSink>(
-      filePath, dwio::common::FileSink::Options{.pool = leafPool_.get()});
+      filePath.string(),
+      dwio::common::FileSink::Options{.pool = leafPool_.get()});
   auto writer = std::make_unique<TextWriter>(
       schema,
       std::move(sink),
       std::make_shared<text::WriterOptions>(writerOptions));
   writer->write(data);
   writer->close();
 
-  const auto fs = filesystems::getFileSystem(tempPath, nullptr);
-  const auto& file = fs->openFileForRead(filePath.string());
-  auto fileSize = file->size();
-
-  BufferPtr charBuf = AlignedBuffer::allocate<char>(fileSize, pool());
-  auto rawCharBuf = charBuf->asMutable<char>();
-  std::vector<std::vector<std::string>> result =
-      parseTextFile(tempPath, filename, rawCharBuf);
-
-  EXPECT_EQ(result.size(), 3);
-  EXPECT_EQ(result[0].size(), 10);
-  // bool type
-  EXPECT_EQ(result[0][0], "true");
-  EXPECT_EQ(result[1][0], "true");
-  EXPECT_EQ(result[2][0], "true");
-
-  // tinyint
-  EXPECT_EQ(result[0][1], "1");
-  EXPECT_EQ(result[1][1], "2");
-  EXPECT_EQ(result[2][1], "3");
-
-  // smallint
-  EXPECT_EQ(result[0][2], "1");
```

Comment:
I would suggest keeping the original test and add your own test of verifying by reader in a new test. 

The reason for that is TextWriter and TextReader could share certain logic or options, if bug was introduced in the common component, then it would be neglected. Having a non-dynamic expected value would be helpful in this case.

If the array printing could be large, in this test, you can use a smaller array or helper function to verify